clean_merchants.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import pandas as pd
+
+def clean_merchants():
+    merchants_df = pd.read_csv('data/unzipped/merchants.csv')
+
+    # There are a lot (over 100000) merchants with incomplete city and state data.
+    # Since it seems unlikely that the location has an influence on the loyalty, we drop these
+    # Also, category_2 is always NaN when the city is unknown, so we drop that, too.
+    merchants_df.drop(columns=['city_id', 'state_id', 'category_2'], inplace=True)
+
+    # There are only 15 merchants where some of the lag data is invalid, so let's just drop these.
+    merchants_df = merchants_df[np.isnan(merchants_df['avg_sales_lag3']) == False]
+
+    # Category_1 and _4 have only values 'N' and 'Y'. Let's replace this with Booleans False and True
```

Comment:
You can also set the fields to be categorical ones (as seen by pandas) in this way:

```python
for v in cat_vars:
    df[v] = df[v].astype('category').cat.as_ordered()
```

Reply:
Interesting. That seems especially useful for categories with 3 and more values,

Reply:
I now changed both categories to categorical data with these three values.
It turns out, for my first step I didn't really need it as all groups take only one of 'Y' or 'N'.

But I am pretty sure that changes when I'm trying to further unify on merchant_id only.